standardtracer/propagation.go

```diff
@@ -44,9 +45,11 @@ func (p *splitTextPropagator) InjectSpan(
 		fieldNameSampled: strconv.FormatBool(sc.raw.Sampled),
 	}
 	sc.Lock()
-	splitTextCarrier.TraceAttributes = make(map[string]string, len(sc.raw.Attributes))
-	for k, v := range sc.raw.Attributes {
-		splitTextCarrier.TraceAttributes[k] = v
+	if l := len(sc.raw.Attributes); l > 0 {
+		splitTextCarrier.TraceAttributes = make(map[string]string, l)
```

Comment:
I think
* if `nil`, create one (if needed, may not actually have any trace attributes)
* if not `nil`, append to provided map

The scenario with `http.Header` is exactly where you need the 2nd case, the tracer only needs to add its data to the headers, not to blank out the existing headers.